skore/src/skore/cross_validate.py

```diff
@@ -0,0 +1,188 @@
+        k: v
+        for k, v in cv_results.items()
+        if not any(added_scorer in k for added_scorer in added_scorers)
+    }
+
+
+def cross_validate(
+    *args, project: Project | None = None, **kwargs
+) -> CrossValidationItem:
+    """Evaluate estimator by cross-validation and output UI-friendly object.
+
+    This function wraps scikit-learn's [cross_validate](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.cross_validate.html#sklearn.model_selection.cross_validate)
+    function, to provide more context
+    and facilitate the analysis.
+    As such, the arguments are the same as scikit-learn's cross_validate function.
+
+    Parameters
+    ----------
+    The same parameters as scikit-learn's cross_validate function, except for
+
+    project : Project or None
+        A project to save cross-validation data into. If None, no save is performed.
+
+    Returns
+    -------
+    CrossValidationItem
+        An object containing the cross-validation results, which can be readily
+        inserted into a Project.
+
+    Examples
+    --------
+    >>> from sklearn import datasets, linear_model
+    >>> diabetes = datasets.load_diabetes()
+    >>> X = diabetes.data[:150]
+    >>> y = diabetes.target[:150]
+    >>> lasso = linear_model.Lasso()
+
+    >>> project = skore.load("project.skore")  # doctest: +SKIP
+    >>> cv_results = cross_validate(lasso, X, y, cv=3, project=project)  # doctest: +SKIP
+    {'fit_time': array(...), 'score_time': array(...), 'test_score': array(...)}
+    """
+    import sklearn.model_selection
+
+    # Extend scorers with other relevant scorers
+
+    # Recover target
+    estimator = args[0] if len(args) >= 1 else kwargs.get("estimator")
+    y = args[2] if len(args) == 3 else kwargs.get("y")
+
+    ml_task = _find_ml_task(estimator, y)
+
+    # Add scorers based on the ML task
+    if ml_task == "clustering":
+        scorers_to_add = ["homogeneity_score", "silhouette_score", "rand_score"]
+    elif ml_task == "regression":
+        scorers_to_add = ["r2", "neg_mean_squared_error"]
+    elif ml_task == "classification":
+        scorers_to_add = ["roc_auc", "neg_brier_score", "recall", "precision"]
+    elif ml_task == "unknown":
+        scorers_to_add = []
+
+    try:
+        scorers = kwargs.pop("scoring")
+    except KeyError:
+        scorers = None
+    new_scorers, added_scorers = _expand_scorers(scorers, scorers_to_add)
+
+    cv_results = sklearn.model_selection.cross_validate(
+        *args, **kwargs, scoring=new_scorers
+    )
+
+    # Recover data
+    X = args[1] if len(args) >= 2 else kwargs.get("X")
+
+    cross_validation_item = CrossValidationItem.factory(cv_results, estimator, X, y)
+
+    if project is not None:
+        project.put_item("cross_validation", cross_validation_item)
+
+    # If in a IPython context (e.g. Jupyter notebook), display the plot
+    try:  # noqa: SIM105
+        display(cross_validation_item.plot)
+    except NameError:
+        pass
```

Comment:
Really disturbing 😅 .

Reply:
So, ruff tells me to use `with contextlib.suppress(NameError)` instead of `try: ... except: pass`, but in this case ruff then complains that `display` is not defined. `display` comes from the `IPython.display` module, so maybe I could replace this with:
```python
with contextlib.suppress(ImportError):
    from IPython.display import display

    display(cross_validation_item.plot)
```

Would that make more sense?

Reply:
Looks better, thanks :) .

Reply:
done